packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/RealmMap.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.types
+
+public interface RealmMap<K, V> : MutableMap<K, V>
+
+public interface RealmDictionary<E> : RealmMap<String, E>
+
+internal class UnmanagedRealmDictionary<E> : RealmDictionary<E>, MutableMap<String, E> by mutableMapOf()
```

Comment:
These should be moved to `io.realm.kotlin.internal` but can be done in a later PR.

Reply:
Yeah, I've moved them in the upcoming PR, I'll leave it as is for now.